sdks/go/pkg/beam/coder_test.go

```diff
@@ -63,3 +64,57 @@ func TestJSONCoder(t *testing.T) {
 		}
 	}
 }
+
+func TestCoders(t *testing.T) {
+	ptrString := "test *string"
+	tests := []interface{}{
+		43,
+		12431235,
+		-2,
+		0,
+		1,
+		true,
+		"a string",
+		map[int64]string{1: "one", 11: "oneone", 21: "twoone", 1211: "onetwooneone"},
+		struct {
+			A int
+			B *string
+			C bool
+		}{4, &ptrString, false},
+		[...]int64{1, 2, 3, 4, 5}, // array
+		[]int64{1, 2, 3, 4, 5},    // slice
+		struct {
+			A []int
+			B [3]int
+		}{A: []int{1, 2, 3}, B: [...]int{4, 5, 6}},
+	}
+
+	for _, test := range tests {
+		var results []string
+		rt := reflect.TypeOf(test)
+		enc := NewElementEncoder(rt)
+		for i := 0; i < 10; i++ {
+			var buf bytes.Buffer
+			if err := enc.Encode(test, &buf); err != nil {
+				t.Fatalf("Failed to encode %v: %v", tests, err)
+			}
+			results = append(results, string(buf.Bytes()))
+		}
+		for i, data := range results {
+			if data != results[0] {
+				t.Errorf("coder not deterministic: data[%d]: %v != %v ", i, data, results[0])
```

Comment:
Nit: data isn't an array but the error message presents it as one. I would have expected results[%d] there.

Reply:
Good catch! Fixed to make it clearer what the test is checking for.